src/tests/unit/cpu/mkldnn_jit_kernel_test.cpp

```diff
@@ -0,0 +1,347 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+#include <utils/jit_kernel.hpp>
+#include <random>
+
+using namespace MKLDNNPlugin;
+using namespace mkldnn::impl::cpu::x64;
+using namespace Xbyak;
+
+#define TEST_JIT_SCALAR_EXPRESSION (c << 5) * b | (a & b - c) | (b - a) >> 2
+
+template<typename Params>
+struct jit_test_kernel : public jit_kernel {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_test_kernel)
+
+    typedef void (*function_t)(const Params *);
+
+    void init() {
+        if (create_kernel() != status::success)
+            IE_THROW() << "Can't generate jit kernel";
+        _fn = (function_t)jit_ker();
+    }
+
+    void operator()(const Params & args) const {
+        _fn(&args);
+    }
+
+private:
+    function_t _fn;
+};
+
+template<typename T>
+struct jit_scalar_variable_test_kernel {
+    struct Params {
+        T a;
+        T b;
+        T c;
+        T d;
+        T *result;
+    };
+
+    void operator()(const Params & args) const {
+        _kernel(args);
+    }
+
+    jit_scalar_variable_test_kernel() {
+        _kernel.init();
+    }
+
+private:
+    class kernel_impl : public jit_test_kernel<Params> {
+        void generate() override {
+            this->preamble();
+
+            auto a = this->arg(&Params::a);
+            auto b = this->arg(&Params::b);
+            auto c = this->arg(&Params::c);
+            auto d = this->arg(&Params::d);
+            auto result = this->arg(&Params::result);
+
+            *result = TEST_JIT_SCALAR_EXPRESSION;
+
+            this->postamble();
+        }
+    };
+
+    kernel_impl _kernel;
+};
+
+TEST(JitKernel, scalar_variable) {
+    size_t a = 1;
+    size_t b = 2;
+    size_t c = 3;
+    size_t d = 4;
+
+    size_t result64 = 0;
+    jit_scalar_variable_test_kernel<size_t> kernel64;
+    jit_scalar_variable_test_kernel<size_t>::Params args64 = { a, b, c, d, &result64 };
+    kernel64(args64);
+
+    uint32_t result32 = 0;
+    jit_scalar_variable_test_kernel<uint32_t> kernel32;
```

Comment:
Is it possible to cover signed and floating point precisions as well or it is not supported by the `jit_kernel` yet?

Reply:
Signed int data types are supported but the single floating point operations aren't implemented.
Added tests for expressions with signed data types.